delegate_main.cc

```diff
@@ -371,22 +371,70 @@ bool Delegate::SupportedOp(TfLiteContext* context,
   return false;
 }
 
-TfLiteDelegate* Delegate::Create() {
-  TfLiteDelegate* delegate = new TfLiteDelegate();
+TfLiteDelegate* Delegate::Create(const VxDelegateOptions* options) {
+  //TfLiteDelegate* delegate = reinterpret_cast<TfLiteDelegate*>(new DerivedDelegateData());
+  DerivedDelegateData* delegate = new DerivedDelegateData();
+  std::memset(delegate, 0, sizeof(DerivedDelegateData));
 
-  std::memset(delegate, 0, sizeof(TfLiteDelegate));
-  delegate->flags = kTfLiteDelegateFlagsNone;
-  delegate->Prepare = &PrepareDelegate;
-  delegate->CopyFromBufferHandle = &CopyFromBufferHandle;
-  delegate->FreeBufferHandle = &FreeBufferHandle;
+  delegate->parent.flags = kTfLiteDelegateFlagsNone;
+  delegate->parent.Prepare = &PrepareDelegate;
+  delegate->parent.CopyFromBufferHandle = &CopyFromBufferHandle;
+  delegate->parent.FreeBufferHandle = &FreeBufferHandle;
 
-  return delegate;
+  delegate->cache_path = options->cache_file_path;
+  delegate->allow_cache_mode = options->allowed_cache_mode;
+  return reinterpret_cast<TfLiteDelegate*>(delegate);
```

Comment:
We also required change the delete part - In C++, delete will call de-constructor according to your data type, to get it cleaned correctly, we need convert data type.